SU2_CFD/src/output/CMultizoneOutput.cpp

```diff
@@ -103,20 +103,11 @@ void CMultizoneOutput::LoadMultizoneHistoryData(const COutput* const* output, co
       const auto& name = item.first;
       nameMultizone = name + zoneIndex;
 
-      /*--- Determine whether nMaker_Analyze/Monitoring has to be looped. ---*/
-      const auto& group = item.second[0].outputGroup;
-      unsigned short nMarker = 0;
-      if (group == "FLOW_COEFF_SURF")
-        nMarker = config[iZone]->GetnMarker_Analyze();
-      else if (group == "AERO_COEFF_SURF")
-        nMarker = config[iZone]->GetnMarker_Monitoring();
-      else
-        SU2_MPI::Error("Per Surface output group unknown: " + group, CURRENT_FUNCTION);
-
-      for (unsigned short iMarker = 0; iMarker < nMarker; iMarker++) {
-        SetHistoryOutputPerSurfaceValue(nameMultizone, item.second[iMarker].value, iMarker);
-      }// for iMarker
-    }// for HistPerSurfFields
+      unsigned short iMarker = 0;
+      for (const auto& field : item.second) {
+        SetHistoryOutputPerSurfaceValue(nameMultizone, field.value, iMarker++);
+      }
```

Comment:
Nice, I knew there was a better way but my smooth brain wouldn't allow me finding a nicer version. This version though is even harder to grasp :D

Reply:
Tobi: "implements"
Pedro: Lets clean that shit up
:smiling_face_with_tear:

Reply:
Why is it hard to grasp? I also missed it in your PR but it's just a vector.

Reply:
That one thing alone of course not, but one needs to dig into how the HistoryOutput works in general and then for multizone things already get quite a bit less obvious than they are for single zone and  PerSurface packs another level of complexity on top.

Reply:
That's true... there is a fair bit of complexity in COutput to allow adding fields easily, still, much much better than legacy.